### PR TITLE
Command channels #566

### LINKED ISSUE #566: Add method to access number of currently-connected exex/htex workers
In other words, the equivalent of calling for example `len(dfk.executors['ipp'].executor.ids)` with the IPP executor. This info is available in the interchange by counting entries in `Interchange._ready_worker_queue`, but for scale testing and user debugging it would be more convenient if that info could be accessed through the executor.

### LINKED ISSUE #574: Write all htex/exex logs to rundir by default

### PR DESCRIPTION
**Pull request by yadudoc**
Fixes #574 : Writing logs to rundir by default
Fixes #566 : Adds command channels to request information from the interchange

### CODE REVIEW

File: parsl/executors/high_throughput/interchange.py [hunk 1]
```diff
@@ -205,20 +247,25 @@ def start(self, poll_period=1):
                                                     args=(self._kill_event,))
         self._task_puller_thread.start()
 
+        logger.debug("Trying to start the command thread")
+        self._command_thread = threading.Thread(target=self._command_server,
+                                                args=(self._kill_event,))
+        self._command_thread.start()
+        logger.debug("Done with starting command thread v02")
```
Comment (annawoodard) [inline parsl/executors/high_throughput/interchange.py]: v02?

File: parsl/tests/manual_tests/htex_local.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            provider=LocalProvider(
+                channel=LocalChannel(),
+                init_blocks=1,
+                max_blocks=1,
+                tasks_per_node=1,  # For HighThroughputExecutor, this option should in most cases be 1
+                launcher=SingleNodeLauncher(),
+            ),
+        )
+    ],
+    strategy=None,
```
Comment (annawoodard) [inline parsl/tests/manual_tests/htex_local.py]: Is it still necessary to disable strategy?

File: parsl/executors/high_throughput/executor.py [hunk 3]
```diff
@@ -137,15 +137,17 @@ def __init__(self,
         self.worker_ports = worker_ports
         self.worker_port_range = worker_port_range
         self.interchange_port_range = interchange_port_range
+        self.run_dir = '.'
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply parsl/executors/high_throughput/executor.py]: The run_dir is not set when the executor is not started by the DFK, for eg when running tests with just the executor.

File: parsl/executors/high_throughput/interchange.py [hunk 4]
```diff
@@ -50,10 +51,11 @@ class Interchange(object):
     def __init__(self,
                  client_address="127.0.0.1",
                  interchange_address="127.0.0.1",
-                 client_ports=(50055, 50056),
+                 client_ports=(50055, 50056, 50057),
                  worker_ports=None,
                  worker_port_range=(54000, 55000),
                  heartbeat_period=60,
+                 logdir=".",
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply parsl/executors/high_throughput/interchange.py]: The interchange and the executor both run (for now) where Parsl runs. The executor starts the interchange with the logdir set to `run_dir`.

File: parsl/executors/high_throughput/executor.py [hunk 5]
```diff
@@ -321,6 +327,24 @@ def _start_queue_management_thread(self):
         else:
             logger.debug("Management thread already exists, returning")
 
+    @property
```
[the remark this answers is not in the document]
Comment (yadudoc) [reply parsl/executors/high_throughput/executor.py]: Not needed. Will remove this.

File: parsl/tests/manual_tests/htex_local.py (hunk 2 above)
Comment (yadudoc) [reply parsl/tests/manual_tests/htex_local.py]: Not necessary at all.

**Pull request merged**